In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

import os

from plot_utils import plot_1Dhist, display_statistics

In [2]:
def read_data(fname, dir="./data/"):
    # Read data from csv file
    return pd.read_csv(os.path.join(dir, fname))

def save_fig(fig, fname, dir="./plots/eda/"):
    # Save figure to file
    fig.savefig(os.path.join(dir, fname), dpi=300, facecolor="w")

In [3]:
PARTICLE_FNAME = "particle_df.csv"

FONTSIZE = 16

SAVE_FIG = False

---

## Particles

In [4]:
# read in the data
par_df = read_data(PARTICLE_FNAME)

# display the first 5 rows of the data
par_df.head()

eventID  jetID  particleType  particleVx  particleVy  particleVz  \
0        0      0             0    0.000000    0.000000    0.000000   
1        0      0             0    0.000000    0.000000    0.000000   
2        0      0          -211   -0.981025    1.422285  -33.456345   
3        0      0           130    0.073932    0.089866   -2.399344   
4        0      0          -211    0.073905    0.089409   -2.399101   

   particlePx  particlePy  particlePz   particleE  particlePolarPx  \
0 -115.595071    5.513218  107.093643  157.675996       115.726471   
1  -83.072377    4.831796   75.798599  112.561324        83.212776   
2  -11.168506   -8.774579    9.043395   16.838385        14.203125   
3   -8.233158   -1.087632    6.647210   10.637351         8.304688   
4   -8.048296    0.478376    6.097900   10.109785         8.062500   

   particlePolarPy  particlePolarPz  particlePolarE  particlePhi  \
0         0.827630         3.093935    2.347607e-01     3.093935   
1         0.816948         3.083494    5.078805e-01     3.083494   
2         0.600055        -2.475661    1.395264e-01    -2.475661   
3         0.732994        -3.010249   -1.192093e-07    -3.010249   
4         0.698202         3.082224    1.395264e-01     3.082224   

   particleTheta  
0       0.824122  
1       0.831991  
2       1.003814  
3       0.895801  
4       0.923257

In [5]:
# Print the number of rows and columns in the DataFrame. 
print("Rows x Columns: ", par_df.shape)

# Print the number of unique events and jets in the DataFrame. 
print("Number of events: ", par_df["eventID"].nunique())
print("Number of jets: ", par_df["jetID"].nunique())

# Print the number of unique particles in the DataFrame.
print("Number of particles: ", par_df["particleType"].nunique())

# List the unique particle types in the DataFrame.
print("Unique particle types: ", par_df["particleType"].unique())

Rows x Columns:  (168058, 16)
Number of events:  3668
Number of jets:  5851
Number of particles:  11
Unique particle types:  [   0 -211  130  211   22    2    1   13  -11   11  -13]


The particle numbering scheme is as follows:

| Number | Particle |
|:------:|:--------:|
| 0      | unknown |
| 1      | down quark |
| 2      | up quark |
| 11     | electron |
| 13     | muon |
| 22     | photon |
| 130    | neutral kaon (long) |
| 211    | charged pion |

where the numbers are the PDG ID numbers. Positive numbers are particles and negative numbers are antiparticles.

